app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java

```diff
@@ -68,21 +73,29 @@ public RedisPlugin(PluginWrapper wrapper) {
                             String.format("Body is null or empty [%s]", query)));
                 }
 
-                // First value will be the redis command and others are arguments for that command
-                String[] bodySplitted = query.trim().split("\\s+");
+                Map cmdAndArgs = getCommandAndArgs(query.trim());
+                if (!isQueryFormatValid(cmdAndArgs)) {
+                    return Mono.error(
+                            new AppsmithPluginException(
+                                    AppsmithPluginError.PLUGIN_EXECUTE_ARGUMENT_ERROR,
+                                    "Appsmith server has failed to parse Redis query. Please check if the Redis query" +
+                                            " has correct format."
```

Comment:
Nit: Minor English grammar correction
```suggestion
                                    "Appsmith server has failed to parse your Redis query. Are you sure it's" +
                                            " been formatted correctly."
```